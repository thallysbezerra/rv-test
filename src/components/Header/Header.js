import React from 'react';

import rvLogo from '../../img/RV-logo.svg';

import './Header.scss';

const Header = () => {
	return (
		<header className="header">
			<div className="header--grid-container container">
				<img
					src={rvLogo}
					className="header__logo"
					alt="Red Ventures - Logo"
				/>
				<nav className="header__nav">
					<ul className="header__nav__ul">
						<li className="header__nav__ul__li">
							<a href="/" className="header__nav__ul__li__a active">
								Model R
							</a>
						</li>
						<li className="header__nav__ul__li">
							<a href="/" className="header__nav__ul__li__a disabled">
								Model IQ
							</a>
						</li>
						<li className="header__nav__ul__li">
							<a href="/" className="header__nav__ul__li__a disabled">
								Model Mobi
							</a>
						</li>
						<li className="header__nav__ul__li">
							<a href="/" className="header__nav__ul__li__a disabled">
								Model Charlie
							</a>
						</li>
						<li className="header__nav__ul__li">
							<a href="/" className="header__nav__ul__li__a disabled">
								Model Italy
							</a>
						</li>
					</ul>
				</nav>
			</div>
		</header>
	);
};

export default Header;
