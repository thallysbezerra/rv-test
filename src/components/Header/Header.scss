@import '../../css/theme';

.header {
	&--grid-container {
		align-items: center;
		box-sizing: border-box;
		display: grid;
		grid-template-columns: 179px 1fr;
		height: 75px;
	}

	&__logo {
		box-sizing: border-box;
		width: 159px;

		@include breakpoint(tabletL) {
			width: 132px;
		}
	}

	&__nav {
		position: relative;

		&:before,
		&:after {
			content: '';
			height: 75px;
			position: absolute;
			top: 0;
			width: 20px;

			@include breakpoint(tabletL) {
				display: none;
			}
		}

		&:before {
			background: linear-gradient(90deg, $white, transparent);
			left: 0;
		}

		&:after {
			background: linear-gradient(90deg, transparent, $white);
			right: 0;
		}

		&__ul {
			align-items: center;
			display: flex;
			height: 75px;
			list-style: none;
			margin: 0;
			max-width: calc(100vw - 219px);
			overflow-x: scroll;
			padding-left: 0;
			white-space: nowrap;

			@include breakpoint(tabletL) {
				justify-content: flex-end;
				overflow-x: hidden;
			}

			&__li {
				display: inline-block;
				margin: 0 25px;

				&__a {
					color: $greyish-blue;
					display: inline-block;
					font-weight: $regular;
					padding: 4px 0 2px;
					text-decoration: none;

					&.active {
						border-bottom: 2px solid $red;
						pointer-events: none;
					}

					&.disabled {
						opacity: 0.57;
						pointer-events: none;
					}
				}

				&:last-of-type {
					padding-right: 25px;

					@include breakpoint(tabletL) {
						margin-right: 0;
						padding-right: 0;
					}
				}
			}
		}
	}
}
