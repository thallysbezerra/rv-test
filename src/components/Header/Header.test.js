import React from 'react';
import Header from './Header';

import renderer from 'react-test-renderer';

describe('Header UI', () => {
	it('<Header /> must have the snapshot version.', () => {
		const tree = renderer.create(<Header />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
