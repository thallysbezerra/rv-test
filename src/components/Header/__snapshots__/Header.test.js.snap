// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI <Header /> must have the snapshot version. 1`] = `
<header
  className="header"
>
  <div
    className="header--grid-container container"
  >
    <img
      alt="Red Ventures - Logo"
      className="header__logo"
      src="RV-logo.svg"
    />
    <nav
      className="header__nav"
    >
      <ul
        className="header__nav__ul"
      >
        <li
          className="header__nav__ul__li"
        >
          <a
            className="header__nav__ul__li__a active"
            href="/"
          >
            Model R
          </a>
        </li>
        <li
          className="header__nav__ul__li"
        >
          <a
            className="header__nav__ul__li__a disabled"
            href="/"
          >
            Model IQ
          </a>
        </li>
        <li
          className="header__nav__ul__li"
        >
          <a
            className="header__nav__ul__li__a disabled"
            href="/"
          >
            Model Mobi
          </a>
        </li>
        <li
          className="header__nav__ul__li"
        >
          <a
            className="header__nav__ul__li__a disabled"
            href="/"
          >
            Model Charlie
          </a>
        </li>
        <li
          className="header__nav__ul__li"
        >
          <a
            className="header__nav__ul__li__a disabled"
            href="/"
          >
            Model Italy
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
